### PR TITLE
Always call into next middleware in WebApplicationBuilder

### PR DESCRIPTION
**Pull request by halter73**
- An IStartupFilter can add middleware later in the pipeline

### CODE REVIEW

File: src/DefaultBuilder/src/WebApplicationBuilder.cs [hunk 1]
```diff
@@ -121,49 +125,46 @@ private void ConfigureApplication(WebHostBuilderContext context, IApplicationBui
...
+            // Wire the source pipeline to run in the destination pipeline
+            app.Use(next =>
+            {
+                _builtApplication.Run(next);
+                return _builtApplication.BuildRequestDelegate();
+            });
+
+            // Implicitly call UseEndpoints() at the end of the pipeline if UseRouting() was called implicitly.
+            // We could add this to the end of _buildApplication instead if UseEndpoints() was not so picky about
+            // being called with the same IApplicationBluilder instance as UseRouting().
```
Comment (davidfowl) [inline src/DefaultBuilder/src/WebApplicationBuilder.cs:164]:
```suggestion
            // being called with the same IApplicationBulilder instance as UseRouting().
```

Comment (davidfowl) [reply src/DefaultBuilder/src/WebApplicationBuilder.cs:164]: This is probably more correct anyways. Is there any observable difference with this ordering?

**Review by davidfowl — APPROVED** [review]

File: src/DefaultBuilder/src/WebApplicationBuilder.cs (hunk 1 above)
Comment (halter73) [reply src/DefaultBuilder/src/WebApplicationBuilder.cs:164]: Nope. I just don't like that it's more lines of code 😄

**Comment by davidfowl** [comment]
Seems like something is actually broken here.

**Pull request merged**